### DISCUSSION TITLE
Unable to access jarfile

### DISCUSSION THREAD

**Discussion by TED-0136** [jagrosh/MusicBot#1203] [Questions] [answered]
As the title, when I try to open the `JMusicBot-0.3.8.jar`, the error message keeps pop up. 
![image](https://user-images.githubusercontent.com/85302244/186725878-e74b38ea-a633-4b20-bd99-d64bbc713223.png)

I've tried to google online "unable to access jarfile", and tried every solution I see,(update my java, download jarfix, etc.), but the problem is still the same. 

To be honest, I am now using my new pc, which I just used for 3 days, my OS is windows 11. 
![image](https://user-images.githubusercontent.com/85302244/186727018-806994e5-0856-49d3-b321-012bd40187c0.png)
The OS of my old laptop was also Windows 11, and I used to put my files on the desktop, but everything worked normally, I never encountered any problem in the past, until now with my new pc. 
I just copied my files to my new pc, but this time I put them in `Documents`, and the problem happened today when I want to use the bot. 

I also tried to delete the files and download a new one, or download the older 0.3.7 release, but both of them also failed to work. 
![image](https://user-images.githubusercontent.com/85302244/186728951-b56fb156-0677-4763-abd8-0e7ccbf6b94a.png)
At the end, I tried to put my files at the destination it said, and it worked again. WHY?_?

Now I can use the bot again, but I still don't get where the problem is, 
and also I wanna ask, is there any way I can move my files to other places without happening the issue?

**Comment by jagrosh** [comment, marked as answer]
It looks like you're trying to run the bot out of OneDrive, which could be the issue. Try running it from a folder within your local Documents folder, _not_ a networked folder.

**Reply by TED-0136** [reply]
Thanks for your reply!

My old laptop doesn't have an OneDrive app, but my new pc somehow has the OneDrive app since the beginning,
and I don't understand how OneDrive works actually. 

Now I tried to unlink my pc with OneDrive, and it works better. But the problem seems not to be solved yet. 
The bot(the jar file) still doesn't work when I put the file in the `Document` folder or `Desktop`. 
But if I put the file in the `Download` folder or `Local Disk (C:)`, it works.

**Reply by TED-0136** [reply]
I made some attempts and found something different. 

If I put the file on the `Desktop`, the error message would look like this: 
![image](https://user-images.githubusercontent.com/85302244/186966006-a9c185a6-236a-4035-b20f-c5a6a40a87de.png)
But if I create a new folder and put my files in it, it would become like this:
![image](https://user-images.githubusercontent.com/85302244/186966132-34f083b3-487f-4d8e-aa28-7b5d6aabbc4c.png)

I guess some folders in my PC are linked with the OneDrive since my OS set up, and some don't, so it's just as you said, don't use a networked folder. 

I think I still have to do some research to figure out how OneDrive actually work with my pc, 
but anyway, thanks for your reply, it helped me a lot!

**Reply by TED-0136** [reply]
Well, I made some different attempts again, and I believe I finally found the real problem.

My guess was wrong, it's probably not OneDrive's fault. 
Here are some of the folders linked with OneDrive in my pc:
![image](https://user-images.githubusercontent.com/85302244/186970783-acf5f121-4c16-4050-967c-afd268b85f14.png)
The weird thing I found is, the bot worked in the `JMusicBot` folder, but not worked in the `Document` folder. 

Until I clicked into the properties of the bot in the `Document` folder and found: The path includes Chinese...............
![image](https://user-images.githubusercontent.com/85302244/186971440-6cff767f-21b3-4790-83ba-91b4a3a46fa5.png)
I want to prevent issues like this so I set the default display language as English from the beginning, when I was setting up my OS, 
but somehow it seems like, there's still something that the system missed. 

Also, now I understand what does the question mark appeared in the error message means
![image](https://user-images.githubusercontent.com/85302244/186972272-5935195e-1e38-4835-8edc-41b3754ba9f9.png)
Those question marks represent every Chinese words. 

My issue is solved, I know how to deal with the bot now.
